typegen/src/tests/mod.rs

```diff
@@ -0,0 +1,1143 @@
+            #[codec(crate = ::subxt_path::ext::codec)]
+            #[decode_as_type(crate_path = ":: subxt_path :: ext :: scale_decode")]
+            #[encode_as_type(crate_path = ":: subxt_path :: ext :: scale_encode")]
+            pub struct TSu16(pub ::core::primitive::u16,);
+
+            #[derive(::subxt_path::ext::codec::CompactAs, ::subxt_path::ext::codec::Decode, ::subxt_path::ext::codec::Encode, ::subxt_path::ext::scale_decode::DecodeAsType, ::subxt_path::ext::scale_encode::EncodeAsType, Debug)]
+            #[codec(crate = ::subxt_path::ext::codec)]
+            #[decode_as_type(crate_path = ":: subxt_path :: ext :: scale_decode")]
+            #[encode_as_type(crate_path = ":: subxt_path :: ext :: scale_encode")]
+            pub struct TSu32(pub ::core::primitive::u32,);
+
+            #[derive(::subxt_path::ext::codec::CompactAs, ::subxt_path::ext::codec::Decode, ::subxt_path::ext::codec::Encode, ::subxt_path::ext::scale_decode::DecodeAsType, ::subxt_path::ext::scale_encode::EncodeAsType, Debug)]
+            #[codec(crate = ::subxt_path::ext::codec)]
+            #[decode_as_type(crate_path = ":: subxt_path :: ext :: scale_decode")]
+            #[encode_as_type(crate_path = ":: subxt_path :: ext :: scale_encode")]
+            pub struct TSu64(pub ::core::primitive::u64,);
+
+            #[derive(::subxt_path::ext::codec::CompactAs, ::subxt_path::ext::codec::Decode, ::subxt_path::ext::codec::Encode, ::subxt_path::ext::scale_decode::DecodeAsType, ::subxt_path::ext::scale_encode::EncodeAsType, Debug)]
+            #[codec(crate = ::subxt_path::ext::codec)]
+            #[decode_as_type(crate_path = ":: subxt_path :: ext :: scale_decode")]
+            #[encode_as_type(crate_path = ":: subxt_path :: ext :: scale_encode")]
+            pub struct TSu8(pub ::core::primitive::u8,);
+        }
+    };
+
+    assert_eq!(code.to_string(), expected_code.to_string());
+}
+
+#[test]
+fn generate_enum() {
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    enum E {
+        A,
+        B(bool),
+        C { a: u32 },
+    }
+
+    let code = Testgen::new().with::<E>().gen_tests_mod(subxt_settings());
+
+    let expected_code = quote! {
+        pub mod tests {
+            use super::root;
+            #[derive(::subxt_path::ext::codec::Decode, ::subxt_path::ext::codec::Encode, ::subxt_path::ext::scale_decode::DecodeAsType, ::subxt_path::ext::scale_encode::EncodeAsType, Debug)]
+            #[codec(crate = ::subxt_path::ext::codec)]
+            #[decode_as_type(crate_path = ":: subxt_path :: ext :: scale_decode")]
+            #[encode_as_type(crate_path = ":: subxt_path :: ext :: scale_encode")]
+            pub enum E {
+                # [codec (index = 0)]
+                A,
+                # [codec (index = 1)]
```

Comment:
Are these inserted by default because we did not specify `insert_codec_attributes: false`? I can't remember off hand, what do we do in subxt by default?

Reply:
Yeah it's good to do it in Subxt so that if we want `Encode` and `Decode`, the indexes are correct (but lovely that you can remove it all in this crate if you don't care)